### PR TITLE
Complete Single Account Mode MSAL for React Native

### PR DESCRIPTION
**Pull request by ahnmelis**
This PR adds the acquireToken function to MSALModule and utilizes it in the sample app. 
So far, the module should be able to wrap MSAL for Android's single account mode functions for AAD, which is the minimum project goal I set for myself. It would be good to get all this polished before I move on to more features, which could include multiple accounts mode.

### CODE REVIEW

File: lib/msal-react-native-android-poc/MSALModule/android/src/main/java/com/reactlibrary/MSALModule.java [hunk 1]
```diff
@@ -175,6 +175,49 @@ public void onError(@NonNull MsalException exception) {
...
+    /**
+     * getAuthInteractiveCallback is the callback function for acquireToken.
+     * If successful, it will return a map of the IAuthenticationResult.
+     * If there's an error, the exception will be returned.
+     */
+
+    private AuthenticationCallback getAuthInteractiveCallback(final Promise promise) {
+        return new AuthenticationCallback() {
+            @Override
+            public void onCancel() {
```
Comment (tnorling) [inline lib/msal-react-native-android-poc/MSALModule/android/src/main/java/com/reactlibrary/MSALModule.java:202]: Is this event raised by MSAL android or the end-user? We have cleanup code for things like cache when a login event is cancelled, you'll want to make sure this is still being done.

File: lib/msal-react-native-android-poc/SampleApp/android/app/src/main/java/com/sampleapp/MSALModule.java [hunk 2]
```diff
@@ -175,6 +175,49 @@ public void onError(@NonNull MsalException exception) {
...
+     * acquireToken: attempts to obtain a token interactively
+     * Parameters: string scopesValue (string containing scopes separated by a " "), Promise promise (returned to an async function)
+     */
+    @ReactMethod
+    public void acquireToken(String scopesValue, Promise promise) {
+        if (!scopesValue.isEmpty()) {
+            publicClientApplication.acquireToken(getCurrentActivity(), scopesValue.toLowerCase().split(" "), getAuthInteractiveCallback(promise));
+        } else {
+            //the default is User.Read
+            String[] array = {"User.Read"};
```
Comment (tnorling) [inline lib/msal-react-native-android-poc/SampleApp/android/app/src/main/java/com/sampleapp/MSALModule.java:188]: This is another spot you could use the constant suggested above

File: lib/msal-react-native-android-poc/MSALModule/android/src/main/java/com/reactlibrary/MSALModule.java [hunk 3]
```diff
@@ -175,6 +175,49 @@ public void onError(@NonNull MsalException exception) {
...
+     * acquireToken: attempts to obtain a token interactively
+     * Parameters: string scopesValue (string containing scopes separated by a " "), Promise promise (returned to an async function)
+     */
+    @ReactMethod
+    public void acquireToken(String scopesValue, Promise promise) {
+        if (!scopesValue.isEmpty()) {
+            publicClientApplication.acquireToken(getCurrentActivity(), scopesValue.toLowerCase().split(" "), getAuthInteractiveCallback(promise));
+        } else {
+            //the default is User.Read
+            String[] array = {"User.Read"};
```
[the remark this answers is not in the document]
Comment (pkanher617) [reply lib/msal-react-native-android-poc/MSALModule/android/src/main/java/com/reactlibrary/MSALModule.java:188]: +1

[the remark this answers is not in the document]
Comment (ahnmelis) [reply lib/msal-react-native-android-poc/MSALModule/android/src/main/java/com/reactlibrary/MSALModule.java:191]: Done! Created Constants.java in the same directory. array is now DEFAULT_SCOPES

File: lib/msal-react-native-android-poc/MSALModule/android/src/main/java/com/reactlibrary/MSALModule.java (hunk 1 above)
Comment (ahnmelis) [reply lib/msal-react-native-android-poc/MSALModule/android/src/main/java/com/reactlibrary/MSALModule.java:205]: I'm waiting from a response from the android team regarding cleanup code, but I believe the event is raised by the end user (when they back out of the auth flow).  Looking at their sample app, all they have within their onCancel is a logging statement. So hopefully I won't have to worry about clearing the cache but I'll check to make sure.

Comment (ahnmelis) [reply lib/msal-react-native-android-poc/MSALModule/android/src/main/java/com/reactlibrary/MSALModule.java:205]: Shane says that MSAL Android doesn't create anything in the token cache until there is an actual token response. So when the user cancels it wouldn't have gotten that far yet

**Pull request merged**